File: PYTHON/Program/Vehicle.py
# Saya Rifa Sania NIM 2206697 mengerjakan Latihan Praktikum 4 dalam mata kuliah Desain Pemrograman Berorientasi Objek
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin...

class Vehicle:
    __platNomor = ""
    __merk = ""
    __tahunProduksi = ""
    __warna = ""

    def __init__(self):
        self.__platNomor = ""
        self.__merk = ""
        self.__tahunProduksi = ""
        self.__warna = ""

    def __init__(self, platNomor = "", merk = "", tahunProduksi = "", warna = ""):
        self.__platNomor = platNomor
        self.__merk = merk
        self.__tahunProduksi = tahunProduksi
        self.__warna = warna    

    def setPlatNomor(self, platNomor):
        self.__platNomor = platNomor

    def getPlatNomor(self):
        return self.__platNomor

    def setMerk(self, merk):
        self.__merk = merk

    def getMerk(self):
        return self.__merk        

    def setTahunProduksi(self, tahunProduksi):
        self.__tahunProduksi = tahunProduksi

    def getTahunProduksi(self):
        return self.__tahunProduksi

    def setWarna(self, warna):
        self.__warna = warna

    def getWarna(self):
        return self.__warna                 

File: CPP/Program/Vehicle.cpp
#pragma once
#include <iostream>
#include <string>

using namespace std;

class Vehicle{
    private:
        string platNomor;
        string merk;
        string tahunProduksi;
        string warna;

    public:
        Vehicle(){
        }

        Vehicle(string platNomor){
            this->platNomor = platNomor;
            merk = "";
            tahunProduksi = "";
            warna = "";
        }

        Vehicle(string plaNomor, string merk){
            this->platNomor = plaNomor;
            this->merk = merk;
            tahunProduksi = "";
            warna = "";
        }

        Vehicle(string platNomor, string merk, string tahunProduksi){
            this->platNomor = platNomor;
            this->merk = merk;
            this->tahunProduksi = tahunProduksi;
            warna = "";
        }
        
        Vehicle(string platNomor, string merk, string tahunProduksi, string warna){
            this->platNomor = platNomor;
            this->merk = merk;
            this->tahunProduksi = tahunProduksi;
            this->warna = warna;
        }

        void setPlatNomor(string platNomor){
            this->platNomor = platNomor;
        }

        string getPlatNomor(){
            return platNomor;
        }

        void setMerk(string merk){
            this->merk = merk;
        }

        string getMerk(){
            return merk;
        }
        
        void setTahunProduksi(string tahunProduksi){
            this->tahunProduksi = tahunProduksi;
        }

        string getTahunProduksi(){
            return tahunProduksi;
        }

        void setWarna(string warna){
            this->warna = warna;
        }

        string getWarna(){
            return warna;
        }

        ~Vehicle(){
        }
};